src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -536,7 +543,8 @@ private protected (Solution oldSolution, Solution newSolution) SetCurrentSolutio
                         continue;
                     }
 
-                    newSolution = newSolution.WithNewWorkspace(oldSolution.WorkspaceKind, oldSolution.WorkspaceVersion + 1, oldSolution.Services);
+                    var newContentVersion = DetermineNextSolutionStateContentVersion(newSolution: newSolution, oldSolution: oldSolution);
+                    newSolution = newSolution.WithNewWorkspace(oldSolution.WorkspaceKind, newContentVersion, oldSolution.Services);
```

Comment:
Are there any cases where WithNewWorkspace doesn't need this logic? Can we just inline the logic into it?